### ISSUE TITLE
EXIF data not complete

### ISSUE THREAD

**Issue by krabina** [SemanticMediaWiki/SemanticExtraSpecialProperties#155]
### Setup and configuration

Component|Version
-- | --
MediaWiki | 1.31.12
PHP | 7.2.24 (fpm-fcgi)
MySQL | 5.5.62-log
ICU | 50.1.2
Semantic Extra Special Properties | 2.1.0
Semantic MediaWiki | 3.1.6

### Issue
It seems that not all EXIF data that is shown in MediaWiki are made available.
E. g. this file https://www.geschichtewiki.wien.gv.at/Datei:Franz_Josefs_Kai_37_-_2021.jpg
shows geographical coordinates

EXIF|Data
-- | --
Geografische Breite | 48° 12′ 51,23″ N
Geografische Länge|16° 22′ 24,6″ E

But these are not shown in the subobject:
https://www.geschichtewiki.wien.gv.at/Spezial:Durchsuchen/:Datei:Franz-20Josefs-20Kai-2037-20-2D-202021.jpg-23-5FEXIFDATA

Ist this just missing here https://github.com/SemanticMediaWiki/SemanticExtraSpecialProperties/blob/master/data/definitions.json
or ist this some other problem?

**Comment by kghbln** [comment]
> Ist this just ...

Yes, I believe this is the only thing that needs to be done. You could try locally.

As far as I remember correctly the geographical information was left out deliberately due to privacy concerns, i.e. it would be good to have a configuration parameter allowing to switch off "sensitive" data. 

One could argue that the data is already in the wiki but making them much easier to retrieve ... However this is probably a whole new issue.

Anyhow, this will indeed be a nice enhancement.

**Comment by krabina** [comment]
I agree that it would be nice to make this configurable. However, I think this is another feature. This here is to take care that at least those displayed by default in MW are being made available:
https://www.mediawiki.org/wiki/MediaWiki:Metadata-fields

Another nice addition would be to store the GPS data as geographical coordinates. Currenly you would have to use templates and queries in order to compile the text coordinates into one coordinate property.

**Comment by krabina** [comment]
For future reference: in order to create a property of type geographic coordinates, probably this has to be done with some kind of function "createCoordinateItemFromExif" similar to 

https://github.com/SemanticMediaWiki/SemanticExtraSpecialProperties/blob/6fc89fbc9a9a7904dc320cbe9b441b86099eac2a/src/PropertyAnnotators/ExifPropertyAnnotator.php#L149

**Comment by alex-mashin** [comment]
> As far as I remember correctly the geographical information was left out deliberately due to privacy concerns, i.e. it would be good to have a configuration parameter allowing to switch off "sensitive" data.
Shouldn't someone concerned about his privacy simply edit or remove EXIF data from the photo he is uploading?

**Comment by krabina** [comment]
Yes this is why I implemented it in the above mentioned PR.
